File: src/main/kotlin/ch/fdlo/hoerbuchspion/webservice/data/ArtistDetails.java
package ch.fdlo.hoerbuchspion.webservice.data;

import javax.persistence.Embeddable;

@Embeddable
public class ArtistDetails {
  private String artistImage = "";
  private int popularity;

  // Required by JPA
  private ArtistDetails() {}

  public void setArtistImage(String artistImage) {
    this.artistImage = artistImage;
  }

  public void setPopularity(int popularity) {
    this.popularity = popularity;
  }
}

File: src/main/kotlin/ch/fdlo/hoerbuchspion/webservice/data/ArtistDetails.kt
package ch.fdlo.hoerbuchspion.webservice.data

import javax.persistence.*

@Embeddable
class ArtistDetails
private constructor() {  // We need a default constructor because of JPA
    @Embedded
    @AttributeOverrides(
        AttributeOverride(name = "large", column = Column(name = "artistImageLarge")),
        AttributeOverride(name = "medium", column = Column(name = "artistImageMedium")),
        AttributeOverride(name = "small", column = Column(name = "artistImageSmall"))
    )
    val artistImage: ImageURLs? = null
    val popularity = 0
}